src/EPR.PRN.ObligationCalculation.Application/Services/ServiceBusProvider.cs

```diff
@@ -12,37 +12,92 @@ public class ServiceBusProvider : IServiceBusProvider
     private readonly ILogger<ServiceBusProvider> _logger;
     private readonly ServiceBusClient _serviceBusClient;
     private readonly ServiceBusConfig _config;
+    private readonly string _logPrefix;
 
     public ServiceBusProvider(ILogger<ServiceBusProvider> logger, ServiceBusClient serviceBusClient, IOptions<ServiceBusConfig> config)
     {
         _logger = logger;
         _serviceBusClient = serviceBusClient;
         _config = config.Value;
+        _logPrefix = nameof(ServiceBusProvider);
     }
-    public async Task SendApprovedSubmissionsToQueue(List<ApprovedSubmissionEntity> approvedSubmissionEntities)
+    public async Task SendApprovedSubmissionsToQueueAsync(List<ApprovedSubmissionEntity> approvedSubmissionEntities)
     {
-        var organisationIds = approvedSubmissionEntities
-                                .Select(r => r.OrganisationId)
-                                .Distinct()
-                                .ToList();
-
-        var sender = _serviceBusClient.CreateSender(_config.QueueName);
-        using ServiceBusMessageBatch messageBatch = await sender.CreateMessageBatchAsync();
-        var options = new JsonSerializerOptions { WriteIndented = true };
-        foreach (var organisationId in organisationIds)
+        try
         {
-            var submissions = approvedSubmissionEntities.Where(s => s.OrganisationId == organisationId).ToList();
-            if (submissions.Count != 0)
+            if (approvedSubmissionEntities.Count == 0)
             {
-                var jsonSumissions = JsonSerializer.Serialize(submissions, options);
-                if (!messageBatch.TryAddMessage(new ServiceBusMessage(jsonSumissions)))
+                _logger.LogInformation("[{LogPrefix}]: No new submissions received from pom endpoint to queue", _logPrefix);
+                return;
+            }
+            var organisationIds = approvedSubmissionEntities
+                                    .Select(r => r.OrganisationId)
+                                    .Distinct()
+                                    .ToList();
+
+            await using var sender = _serviceBusClient.CreateSender(_config.ObligationQueueName);
+            using ServiceBusMessageBatch messageBatch = await sender.CreateMessageBatchAsync();
+            var options = new JsonSerializerOptions { WriteIndented = true };
+            foreach (var organisationId in organisationIds)
+            {
+                var submissions = approvedSubmissionEntities.Where(s => s.OrganisationId == organisationId).ToList();
+                if (submissions.Count != 0)
```

Comment:
Can this check not be reversed:

if (submissions.Count == 0 )
 Continue;

It would make the flow a little easier to read.